modules/@apostrophecms/schema/ui/apos/components/AposInputSlug.vue

```diff
@@ -45,29 +31,14 @@ export default {
   computed: {
     tabindex () {
       return this.field.disableFocus ? '-1' : '0';
+    }
+  },
+  watch: {
+    next(value) {
+      this.checkTaken();
     },
-    type () {
-      if (this.field.type) {
-        return this.field.type;
-      } else {
-        return 'text';
-      }
-    },
-    classes () {
-      return [ 'apos-input', 'apos-input--text' ];
-    },
-    icon () {
-      if (this.error) {
-        return 'circle-medium-icon';
-      } else if (this.field.type === 'date') {
-        return 'calendar-icon';
-      } else if (this.field.type === 'time') {
-        return 'clock-icon';
-      } else if (this.field.icon) {
-        return this.field.icon;
-      } else {
-        return null;
-      }
+    followsValue(oldVal, newVal) {
+      this.next = newVal ? slugify(newVal) : '';
```

Comment:
When the component is first initialized it should check whether the existing value of the last part (after the last /, if any) of `slug` is in harmony with the existing value of `title` (the slugification of title equals slug).

If it does, then a flag should be set to *not* track changes in the title.

However, the `followsValue` watcher can still keep an eye to see if the title comes back into harmony with the slug. If it does, the flag can be un-set, allowing tracking to begin again.

Otherwise, any change to title will ruin a previously intentionally shortened slug.

Reply:
We also don't want to update the slug value if it's been set manually, right? If I set the slug myself I don't want it changing simply because I later change the piece title.